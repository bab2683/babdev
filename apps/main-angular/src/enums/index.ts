export * from './contact-link.enum';
export * from './languages.enum';
