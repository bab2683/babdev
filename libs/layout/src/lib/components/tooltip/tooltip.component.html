<div #tooltip="matTooltip" [matTooltip]="key | translate | async"></div>
