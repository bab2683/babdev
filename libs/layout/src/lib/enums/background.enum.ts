export enum BackGroundPositionXEnum {
  LEFT = 'left',
  CENTER = 'center',
  RIGHT = 'right'
}

export enum BackGroundPositionYEnum {
  BOTTOM = 'bottom',
  CENTER = 'center',
  TOP = 'top'
}
