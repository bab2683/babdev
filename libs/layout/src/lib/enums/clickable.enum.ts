export enum ClickableActions {
  COPY = 'copy',
  HYPERLINK = 'hyperlink',
  MAIL = 'mail'
}
