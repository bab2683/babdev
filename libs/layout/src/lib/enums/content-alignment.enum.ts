export enum HorizontalContentAlignmentEnum {
  CENTER = 'center',
  LEFT = 'flex-end',
  RIGHT = 'flex-start'
}

export enum VerticalContentAlignmentEnum {
  BOTTOM = 'bottom',
  CENTER = 'center',
  TOP = 'top'
}
