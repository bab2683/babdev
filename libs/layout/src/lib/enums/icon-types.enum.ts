export enum IconTypesEnum {
  FAB = 'fab',
  FAS = 'fas'
}
