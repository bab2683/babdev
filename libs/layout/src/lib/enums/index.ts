export * from './background.enum';
export * from './clickable.enum';
export * from './icon-types.enum';
export * from './tooltip.enum';
