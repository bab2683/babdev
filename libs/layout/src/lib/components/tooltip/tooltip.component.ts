import { Component, Input, ViewChild } from '@angular/core';
import { MatTooltip } from '@angular/material/tooltip';

import { TooltipStatus } from '../../enum';

@Component({
  selector: 'babdev-tooltip',
  templateUrl: './tooltip.component.html',
  styleUrls: ['./tooltip.component.scss']
})
export class TooltipComponent {
  @Input() public key: string;
  @ViewChild('tooltip', { static: false }) public tooltip: MatTooltip;

  public status: TooltipStatus = TooltipStatus.CLOSED;

  public show(duration?: number): void {
    if (this.status === TooltipStatus.CLOSED) {
      this.tooltip.show();
      this.status = TooltipStatus.OPEN;

      if (duration) {
        setTimeout(() => {
          this.hide();
        }, duration);
      }
    }
  }

  public hide(): void {
    if (this.status === TooltipStatus.OPEN) {
      this.tooltip.hide();
      this.status = TooltipStatus.CLOSED;
    }
  }
}
