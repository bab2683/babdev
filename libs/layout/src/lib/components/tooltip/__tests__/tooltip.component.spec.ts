import { Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { MatTooltip, MatTooltipModule } from '@angular/material/tooltip';
import { TranslatePipeMock } from '@babdev/translate-testing';

import { TooltipStatus } from '../../../enums';
import { TooltipComponent } from '../tooltip.component';

@Component({
  template: ''
})
class MatTooltipStubComponent {
  public show = jest.fn();
  public hide = jest.fn();
}

describe('TooltipComponent', () => {
  let component: TooltipComponent;
  let fixture: ComponentFixture<TooltipComponent>;
  let tooltipShowSpy: any;
  let tooltipHideSpy: any;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TooltipComponent, MatTooltipStubComponent, TranslatePipeMock],
      imports: [MatTooltipModule]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TooltipComponent);
    component = fixture.componentInstance;

    component.tooltip = TestBed.createComponent<MatTooltip>(
      MatTooltipStubComponent as any
    ).componentInstance;

    tooltipShowSpy = jest.spyOn(component.tooltip, 'show');
    tooltipHideSpy = jest.spyOn(component.tooltip, 'hide');
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  describe('tooltip hide', () => {
    it('should hide the tooltip', () => {
      component.status = TooltipStatus.OPEN;
      component.hide();
      expect(tooltipHideSpy).toHaveBeenCalled();
    });

    it('should not call matTooltip hide if already closed', () => {
      component.hide();
      expect(tooltipHideSpy).not.toHaveBeenCalled();
    });
  });

  describe('tooltip show', () => {
    beforeEach(() => {
      component.status = TooltipStatus.CLOSED;
    });

    it('should call matTooltip show function', () => {
      component.show();
      expect(component.status).toEqual(TooltipStatus.OPEN);
      expect(tooltipShowSpy).toHaveBeenCalled();
    });

    it('should not call matTooltip show when already open', () => {
      component.status = TooltipStatus.OPEN;
      component.show();
      expect(tooltipShowSpy).not.toHaveBeenCalled();
    });

    it('should call matTooltip show function with auto closing', () => {
      jest.useFakeTimers();
      const timerDuration: number = 50;
      component.show(timerDuration);
      expect(tooltipShowSpy).toHaveBeenCalled();
      expect(setTimeout).toHaveBeenCalledTimes(1);
      expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), timerDuration);
      jest.advanceTimersByTime(timerDuration + 1);
      expect(tooltipHideSpy).toHaveBeenCalled();
    });
  });
});
