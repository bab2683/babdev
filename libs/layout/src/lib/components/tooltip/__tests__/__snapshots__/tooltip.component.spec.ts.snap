// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipComponent should match snapshot 1`] = `
<babdev-tooltip
  status={[Function String]}
  tooltip={[Function MatTooltip]}
>
  <div />
</babdev-tooltip>
`;
